c7n/resources/account.py

```diff
@@ -510,6 +511,18 @@ def process(self, resources, event=None):
         checks = client.describe_trusted_advisor_check_result(
             checkId=self.check_id, language='en')['result']
 
+        # Check status and if necessary refresh checks
+        if checks['status'] == 'not_available':
```

Comment:
we'll need to pull this into a static method (wait_for_check) to enable better testing on it.

Reply:
I'd be happy to help but I don't know exactly what you need done....

Reply:
nutshell, just looking to get some unit tests for this, but since its an exceptional condition, we'll need to mock it. I'll work on it, I'd like to get this in the next release.